code/backend/batch/utilities/helpers/llm_helper.py

```diff
@@ -154,3 +156,34 @@ def get_sk_service_settings(self, service: AzureChatCompletion):
                 max_tokens=self.llm_max_tokens,
             ),
         )
+
+    def get_ml_client(self):
+        if not hasattr(self, "_ml_client"):
+            self._ml_client = MLClient(
+                DefaultAzureCredential(),
```

Comment:
I think this is RBAC authentication, which isn't fully supported yet. Should we change this to keys instead?

I _think_ you can use `AzureKeyCredential` instead.

Reply:
discussed offline. ml client does not support keys as aml workspaces do not use keys. using key to auth to endpoint would require changing the invoke method to generic python sdk and getting endpoint key in script and uploading that to key vault. decided to keep the current auth method as hybrid of keys and rbac (using rbac to list keys).